code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/edge/integration/util/TestSetupHelper.kt

```diff
@@ -11,38 +11,52 @@
 
 package com.adobe.marketing.mobile.edge.integration.util
 
+import com.adobe.marketing.mobile.Edge
 import com.adobe.marketing.mobile.Event
+import com.adobe.marketing.mobile.edge.integration.BuildConfig
 import com.adobe.marketing.mobile.util.TestConstants
 import com.adobe.marketing.mobile.util.TestHelper
 
 object TestSetupHelper {
     /**
-     * Returns the environment file ID for the provided edgeEnvironment value.
+     * Retrieves the Edge location hint from the shell environment.
      *
-     * @param edgeEnvironment The edgeEnvironment value to use.
-     * @return The environment file ID for the provided edgeEnvironment value.
+     * @return The Edge location hint if set in the environment, or `""` if not set.
      */
-    fun getEnvironmentFileID(edgeEnvironment: String): String {
-        when (edgeEnvironment) {
-            "prod" -> {
-                return "94f571f308d5/6b1be84da76a/launch-023a1b64f561-development"
-            }
-
-            "pre-prod" -> {
-                return "94f571f308d5/6b1be84da76a/launch-023a1b64f561-development"
+    val defaultLocationHint: String?
+        get() {
+            return when (val locationHint = BuildConfig.EDGE_LOCATION_HINT) {
+                IntegrationTestConstants.LocationHintMapping.EMPTY_STRING -> ""
```

Comment:
nit: reorder 

IntegrationTestConstants.LocationHintMapping.NONE -> null
IntegrationTestConstants.LocationHintMapping.EMPTY_STRING -> ""
IntegrationTestConstants.LocationHintMapping.INVALID -> locationHint
else -> locationHint

Also we can remove the INVALID case and it can be handled in else unless you want to keep it for possible change of behavior in future.